### PR TITLE
Add build staging command with FIRE_ENV variable

### PR DESCRIPTION
**Pull request by matty-charlton**
- Add a build staging command with `FIRE_ENV=bootbag-staging` set, if no command is set it'll fallback to `bootbag-production`
- Add a `FUNCTIONS_BASE_URL` env variable with existing `process.env.FIRE_ENV` and use within UserService
- Add the correct `nuxt start` command instead of using the dev server for a local production server

When building using `npm run build:staging` it sets the endpoint to staging:
Screenshot 2023-03-14 at 16 37 12

When building using `npm run build` it sets the endpoint to production:
Screenshot 2023-03-14 at 16 39 32

### CODE REVIEW

**Review by FlamurMMT — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -9,7 +9,8 @@
     "css-watch": "npm run css-build -- --watch",
     "dev": "cross-env NODE_ENV=development nodemon server/index.js --watch server",
     "build": "nuxt generate",
-    "start": "cross-env NODE_ENV=production node server/index.js",
+    "build:staging": "FIRE_ENV=bootbag-staging nuxt generate",
+    "start": "nuxt start",
```
Comment (DaisyyWard) [inline package.json:13]: Just curious why the start command changed?

File: services/user-service.js [hunk 2]
```diff
@@ -3,10 +3,7 @@ export class UserService {
     try {
       if (email.trim() !== '') {
         const payload = { email }
-        const uri =
-          process.env.NODE_ENV !== 'production'
-            ? process.env.PRE_REG_STAGING_URI
-            : process.env.PRE_REG_PRODUCTION_URI
+        const uri = `${process.env.FUNCTIONS_BASE_URL}/api-http-public-preRegistration`
```
Comment (DaisyyWard) [inline services/user-service.js:6]: I don't know the code base well at all but is it common for the `/api-http...` but it be in this file or do we set it in a different file and reference it?

File: package.json (hunk 1 above)
Comment (matty-charlton) [reply package.json:13]: The previous `npm run start` command doesn't actually work (just returns a server error once the port is opened), so I've switched it to use the `nuxt start` which is the standard nuxt command - https://nuxtjs.org/docs/get-started/commands/

Comment (DaisyyWard) [reply package.json:13]: Okay cool, I didn't know it didn't work :)

File: services/user-service.js (hunk 2 above)
Comment (matty-charlton) [reply services/user-service.js:6]: Sounds good, will do this :)

Comment (matty-charlton) [reply services/user-service.js]: @DaisyyWard updated to use constants file

**Review by DaisyyWard — APPROVED** [review]

**Pull request merged**